src/gh_pr_upsert/core.py

```diff
@@ -1,2 +1,68 @@
-def hello_world():
-    return "Hello, world!"
+from gh_pr_upsert import git
+from gh_pr_upsert.exceptions import NoChangesError, OtherPeopleError, SameBranchError
+
+
+def pr_upsert(
+    base_repo,
+    base_branch,
+    local_branch,
+    head_repo,
+    head_branch,
+    title,
+    body,
+    close_comment,
+):  # pylint:disable=too-many-arguments
```

Comment:
We could probably code-design around the pylint warning here but again it's probably not worth the effort for this little automation tool.

For what it's worth I suspect we have some missing domain model objects here. I think we could have a `Branch` model with `name` (`str`) and `remote` attrs where `remote` would be `None` for local branches and would be an instance of a new `Remote` model for remote branches. The new `Remote` model would have `name` (`str`) and `github_repo` (`GitHubRepo`) attrs.

Then it'd look something like this:

```python
def pr_upsert(local_branch, head_branch, base_branch, title, body, close_comment):
    ...
```

Usage would be something like:

```python
local_branch = git.current_branch()  # Returns a Branch object with remote=None.
head_branch = Branch.get(remote="origin", name=local_branch.name)
base_branch = Branch.get(remote="origin", name=head_branch.remote.github_repo.default_branch)

pr_upsert(local_branch, head_branch, base_branch, title, body, close_comment)
```